core/app/mailers/spree/promotion_code_batch_mailer.rb

```diff
@@ -0,0 +1,31 @@
+module Spree
+  class PromotionCodeBatchMailer < ApplicationMailer
+    def promotion_code_batch_finished(promotion_code_batch)
+      mail(
+        to: promotion_code_batch.email,
+        body: Spree.t(
+          "promotion_code_batches.finished",
+          number_of_codes: promotion_code_batch.number_of_codes
+        ),
+        subject: Spree.t(
+          "promotion_code_batches.email_subject.batch_finished",
+          promotion_code_batch_id: promotion_code_batch.id
+        ),
+      )
+    end
+
+    def promotion_code_batch_errored(promotion_code_batch)
+      mail(
+        to: promotion_code_batch.email,
+        body: Spree.t(
+        "promotion_code_batches.errored",
+          error: promotion_code_batch.error
+        ),
+        subject: Spree.t(
```

Comment:
Ok, you are passing the id in here. That's why you need you own translation scope. 

Hmm, I doubt that any user has a real understanding where this id is coming from and what to do with it.

Maybe a link to the failed batch in the admin would be much more useful. And this correlates to my former comment: Let's use mailer views instead of a hard coded body.